src/content/language/pattern-types.md

```diff
@@ -438,27 +438,39 @@ If an object has extra fields that the pattern doesn't destructure, it can still
 
 ## Wildcard
 
-`_`
+`_`  
+`<type>()`
 
-A pattern named `_` is a wildcard, either a [variable pattern](#variable) or
-[identifier pattern](#identifier), that doesn't bind or assign to any variable.
+Wildcard patterns don't bind or assign to any variable.
 
-It's useful as a placeholder in places where you need a subpattern in order to
-destructure later positional values:
+These patterns can either be a [variable pattern](#variable) (`_`) or
+[identifier pattern](#identifier) (`<type>()`).
+
+* Object patterns are more concise and convenient,
+  even more so with generic types.
+  With generic types, an object pattern (`MyType()`) infers the
+  most specific type arguments from the matched value.
+* A variable pattern (`MyType _`) uses less specific, often dynamic,
+  type arguments.
+* With non-generic types, there's no reason to prefer variable patterns.
+
+When you need a subpattern to destructure later positional values,
+use a wildcard as a placeholder. This example uses a variable pattern.
 
 <?code-excerpt "language/lib/patterns/pattern_types.dart (wildcard)"?>
 ```dart
 var list = [1, 2, 3];
 var [_, two, _] = list;
 ```
 
-A wildcard name with a type annotation is useful when you want to test a value's
-type but not bind the value to a name:
+To test a value's type but not bind the value to a name,
+use a wildcard name with a type annotation.
+This example uses an object pattern.
 
 <?code-excerpt "language/lib/patterns/pattern_types.dart (wildcard-typed)"?>
 ```dart
 switch (record) {
-  case (int _, String _):
+  case (int(), String()):
     print('First field is int and second is String.');
 }
 ```
```

Comment:
Again, I think it would be OK to keep lines 455-464 from the old text unchanged.

What I wanted to propose in #5926 was actually that there should be an extra paragraph here at the end of this section, perhaps something like the following:

--------------------------------------------------------

This approach works fine for non-generic types like `int` and `String`. However, if you wish to test whether the matched value has a generic type then you need to specify the type arguments explicitly. The section about object patterns shows how you can use object patterns to infer the type arguments, which may be both more convenient and safer. Here is an example:

<?code-excerpt "some magic ;-)"?>
```dart
Iterable<int> iter = [2, 3, 4];
switch (iter) {
  case List<int> _:
    print('It was a list whose first element is ${iter[0].isEven}');
}
```

If we had used `case List _:` rather than `case List<int> _:` then it means `case List<dynamic> _:`, which in turn means that `iter` does not get promoted, and `iter[0]` is a compile-time error.

--------------------------------------------------------

Then we could have something like the following at the end of the 'Object' section:

--------------------------------------------------------

An extreme example is the case where the object pattern doesn't specify any properties at all. In this case the object pattern amounts to a type test. It is particularly useful in the case where the given type is generic, because the actual type arguments are inferred from the static type of the matched value. For example:

<?code-excerpt "some magic ;-)"?>
```dart
Iterable<int> iter = [2, 3, 4];
switch (iter) {
  case List():
    print('It was a list whose first element is ${iter[0].isEven}');
}
```

The object pattern `List()` is inferred to mean `List<int>()` because the static type of `iter` is `Iterable<int>`, and this means that the invocation of `iter[0].isEven` is type correct.

Hence, it may be a good habit to use an "empty" object pattern like `List()` when you just want to test the type of the matched value, rather than using a wildcard like `List<int> _`.

--------------------------------------------------------